compiler/noirc_frontend/src/ast/mod.rs

```diff
@@ -553,6 +602,17 @@ impl UnresolvedTypeExpression {
             }
         }
     }
+
+    pub(crate) fn is_valid_expression(&self) -> bool {
+        match self {
+            UnresolvedTypeExpression::Variable(path) => path.is_ident(),
```

Comment:
The comment in elaborator/mod.rs mentions this function should allow globals & constants. We can have multiple segment paths that still resolve to constants or globals. E.g. `crate::CONSTANT` or `foo::bar::my_global`, but this method disallows these.